src/ports/trending/component.ts

```diff
@@ -27,14 +27,13 @@ export function createTrendingsComponent(
       sortBy: SaleSortBy.MOST_EXPENSIVE, // sort by volume
     })
 
-    // Fetch all the nfts from those 24hs sales
-    const nftResults = (
+    // Fetch all the items from those 24hs sales
+    const items = (
       await Promise.all(
         sales.map((sale) =>
```

Comment:
I think this is going to include repeated items in the list, and do unnecessary fetchs. For instance if there are 20 sales in the last 24hs for the the same item of the same collection there are going to be 20 fetchs for that same item. 

If we moved the computation of the `trendingSales` record that's below this block to happen first, then we could use the keys of that record to fetch the required items, like this:

```ts
// First compute the trending sales record
const trendingSales = ...

// Then use the keys to fetch the items
const items = Promise.all(
  Object.keys(trendingSales).map(key => {
    const [contractAddress, itemId] = key.split('-')
    return itemsComponent.fetch({ contractAddress, itemId })
  })
).recuce(a, b) => a.concat(b), []) // flatten
```


```

Reply:
Ohh nice catch! yeah, it can definitely be moved after to avoid those extra fetches. Updated!